src/Tools/dotnet-gcdump/CommandLine/CollectCommandHandler.cs

```diff
@@ -39,53 +39,22 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                     Console.Out.WriteLine($"-p|--process-id is required");
                     return -1;
                 }
+                
+                var (success, _) = await CollectGCDump(ct, processId, output, timeout, verbose, true);
 
-                output = string.IsNullOrEmpty(output) ? 
-                    $"{DateTime.Now.ToString(@"yyyyMMdd\_hhmmss")}_{processId}.gcdump" :
-                    output;
-
-                FileInfo outputFileInfo = new FileInfo(output);
-
-                if (outputFileInfo.Exists)
-                {
-                    outputFileInfo.Delete();
-                }
-
-                if (string.IsNullOrEmpty(outputFileInfo.Extension) || outputFileInfo.Extension != ".gcdump")
-                {
-                    outputFileInfo = new FileInfo(outputFileInfo.FullName + ".gcdump");
-                }
-
-                Console.Out.WriteLine($"Writing gcdump to '{outputFileInfo.FullName}'...");
-                var dumpTask = Task.Run(() => 
+                if (success)
                 {
-                    var memoryGraph = new Graphs.MemoryGraph(50_000);
-                    var heapInfo = new DotNetHeapInfo();
-                    if (!EventPipeDotNetHeapDumper.DumpFromEventPipe(ct, processId, memoryGraph, verbose ? Console.Out : TextWriter.Null, timeout, heapInfo))
-                        return false;
-                    memoryGraph.AllowReading();
-                    GCHeapDump.WriteMemoryGraph(memoryGraph, outputFileInfo.FullName, "dotnet-gcdump");
-                    return true;
-                });
-
-                var fDumpSuccess = await dumpTask;
-
-                if (fDumpSuccess)
-                {
-                    outputFileInfo.Refresh();
-                    Console.Out.WriteLine($"\tFinished writing {outputFileInfo.Length} bytes.");
                     return 0;
                 }
-                else if (ct.IsCancellationRequested)
+
+                if (ct.IsCancellationRequested)
```

Comment:
Per my comment above, I don't think this will be hit when the task is cancelled and instead will manifest as an `AggregateException` with a `TaskCanceledException` inside it.  [It's worth checking out the docs on `Async` exceptions](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/exception-handling-task-parallel-library#exceptions-that-indicate-cooperative-cancellation) and updating the cath below to handle the `AggregateException`